crates/worktree/src/worktree.rs

```diff
@@ -3283,7 +3283,7 @@ impl Entry {
             path,
             inode: metadata.inode,
             mtime: Some(metadata.mtime),
-            is_symlink: metadata.is_symlink,
+            canonical_path: None,
```

Comment:
That does not look right, does it?
What if we actually ask for a canonical path in the constructor?

Reply:
I was a bit hesitant/nervous changing too much but yeah its true. I will adjust this  :+1: